zipline/pipeline/loaders/utils.py

```diff
@@ -60,70 +60,39 @@ def next_date_frame(dates, events_by_sid, event_date_field_name):
     return pd.DataFrame(index=dates, data=cols)
 
 
-def previous_date_frame(date_index, events_by_sid, event_date_field_name):
+def get_previous_date_or_value(events_by_sid, date_index, missing_value,
```

Comment:
Then we should probably also change `next_date_frame` to be `next_event_frame`, no? And change the functionality of that method to allow handling values that are not dates.